bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetEnergyHandler.java

```diff
@@ -0,0 +1,128 @@
+import org.openwebnet4j.message.Who;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link OpenWebNetEnergyHandler} is responsible for handling commands/messages for a Energy Management OpenWebNet
+ * device. It extends the abstract {@link OpenWebNetThingHandler}.
+ *
+ * @author Massimo Valla - Initial contribution
+ * @author Andrea Conte - Energy management
+ */
+@NonNullByDefault
+public class OpenWebNetEnergyHandler extends OpenWebNetThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(OpenWebNetEnergyHandler.class);
+
+    public final static Set<ThingTypeUID> SUPPORTED_THING_TYPES = OpenWebNetBindingConstants.ENERGY_MANAGEMENT_SUPPORTED_THING_TYPES;
+
+    public OpenWebNetEnergyHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    protected Where buildBusWhere(String wStr) throws IllegalArgumentException {
+        return new WhereEnergyManagement(wStr);
+    }
+
+    @Override
+    public void initialize() {
+        logger.debug("initialize() thing={}", thing.getUID());
+        super.initialize();
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("dispose() thing={}", thing.getUID());
+        super.dispose();
+    }
+
+    @Override
+    protected void requestChannelState(ChannelUID channel) {
+        logger.debug("requestChannelState() thingUID={} channel={}", thing.getUID(), channel.getId());
+        try {
+            bridgeHandler.gateway.send(EnergyManagement.requestActivePower(deviceWhere.value()));
+        } catch (OWNException e) {
+            logger.error("requestChannelState() OWNException thingUID={} channel={}: {}", thing.getUID(),
+                    channel.getId(), e.getMessage());
+        }
+    }
+
+    @Override
+    protected void handleChannelCommand(ChannelUID channel, Command command) {
+        logger.warn("handleChannelCommand() Read only channel, unsupported command {}", command);
+    }
+
+    @Override
+    protected String ownIdPrefix() {
+        return Who.ENERGY_MANAGEMENT.value().toString();
+    }
+
+    @Override
+    protected void handleMessage(BaseOpenMessage msg) {
+        logger.debug("handleMessage({}) for thing: {}", msg, thing.getUID());
+        super.handleMessage(msg);
+
+        if (msg.isCommand()) {
+            logger.warn("handleMessage() Ignoring unsupported command for thing {}. Frame={}", getThing().getUID(),
+                    msg);
+            return;
+        } else {
+            updateActivePower(msg);
+        }
+    }
+
+    /**
+     * Updates energy power state based on a EnergyManagement message received from the OWN network
+     *
+     * @param msg the EnergyManagement message received
+     * @throws FrameException
+     */
+    // private void updateActivePower(EnergyManagement msg) {
+    private void updateActivePower(BaseOpenMessage msg) {
+        logger.debug("updateActivePower() for thing: {}", thing.getUID());
```

Comment:
Can this message be replaced by using the debugger or by increasing the framework's log level? See point 4 https://www.openhab.org/docs/developer/guidelines.html#f-logging

Reply:
removed